src/overlays/actors/ovl_En_Skj/z_en_skj.c

```diff
@@ -39,201 +180,1498 @@ static ColliderCylinderInitType1 sCylinderInit = {
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 0xA, this->unk_2F0, 0);
+    Math_ApproachF(&this->unk_2F0, 2000.0f, 1.0f, 100.0f);
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    if (this->actor.xzDistToPlayer < 120.0f) {
+        this->actor.speedXZ = 0.0f;
+        EnSkj_SetupAskForMask(this, globalCtx);
+    }
+}
+
+void EnSkj_SetupAskForMask(EnSkj* this, GlobalContext* globalCtx) {
+    func_8010B680(globalCtx, 0x101C, &this->actor);
+    EnSkj_ChangeAnim(this, SKJ_ANIM_WAIT);
+    EnSkj_SetupAction(this, SKJ_ACTION_SARIA_SONG_ASK_FOR_MASK);
+}
+
+void EnSkj_AskForMask(EnSkj* this, GlobalContext* globalCtx) {
+    if (func_8010BDBC(&globalCtx->msgCtx) == 4) {
+        if (func_80106BC8(globalCtx) != 0) {
+            switch (globalCtx->msgCtx.choiceIndex) {
+                case 0: // Yes
+                    EnSkj_SetupTakeMask(this, globalCtx);
+                    break;
+                case 1: // No
+                    func_8010B720(globalCtx, 0x101D);
+                    EnSkj_SetupWaitForMaskTextClear(this);
+                    break;
+            }
+        }
+    }
+}
+
+void EnSkj_SetupTakeMask(EnSkj* this, GlobalContext* globalCtx) {
+    func_8010B720(globalCtx, 0x101E);
+    EnSkj_SetupAction(this, SKJ_ACTION_SARIA_SONG_TAKE_MASK);
+}
+
+void EnSkj_TakeMask(EnSkj* this, GlobalContext* globalCtx) {
+    if ((func_8010BDBC(&globalCtx->msgCtx) == 6) && (func_80106BC8(globalCtx) != 0)) {
+        Rupees_ChangeBy(10);
+        gSaveContext.itemGetInf[3] |= 0x200;
+        EnSkj_SetNaviId(this);
+        Player_UnsetMask(globalCtx);
+        Item_Give(globalCtx, ITEM_SOLD_OUT);
+        func_8010B720(globalCtx, 0x101F);
+        EnSkj_SetupWaitForMaskTextClear(this);
+    }
+}
+
+void EnSkj_SetupWaitForMaskTextClear(EnSkj* this) {
+    EnSkj_SetupAction(this, SKJ_ACTION_SARIA_SONG_WAIT_MASK_TEXT);
+}
+
+void EnSkj_WaitForMaskTextClear(EnSkj* this, GlobalContext* globalCtx) {
+    if ((func_8010BDBC(&globalCtx->msgCtx) == 6) && (func_80106BC8(globalCtx) != 0)) {
+        func_8002DF54(globalCtx, &this->actor, 7);
+        this->backfilpFlag = 1;
+        EnSkj_Backflip(this);
+    }
+}
+
+void EnSkj_SetupWrongSong(EnSkj* this) {
+    this->textId = 0x1041;
+    EnSkj_ChangeAnim(this, SKJ_ANIM_WAIT);
+    EnSkj_SetupAction(this, SKJ_ACTION_SARIA_SONG_WRONG_SONG);
+}
+
+void EnSkj_WrongSong(EnSkj* this, GlobalContext* globalCtx) {
+    if (D_80B01EA0 != 0) {
+        EnSkj_SetupWaitForTextClear(this);
+    } else {
+        func_8002F2CC(&this->actor, globalCtx, EnSkj_GetItemXzRange(this));
+    }
+}
+
+void EnSkj_SetupWaitForTextClear(EnSkj* this) {
+    EnSkj_SetupAction(this, SKJ_ACTION_SARIA_SONG_WAIT_FOR_TEXT);
+}
+
+void EnSkj_SariasSongWaitForTextClear(EnSkj* this, GlobalContext* globalCtx) {
+    u8 state;
+    Player* player;
+
+    if (((state = func_8010BDBC(&globalCtx->msgCtx)), player = PLAYER, state == 6) && func_80106BC8(globalCtx)) {
```

Comment:
By any chance, can the state and player initializations be moved outside of the if here?